Storage/SQL/SQLiteBookmarks.swift

```diff
@@ -197,33 +197,38 @@ public class SQLiteBookmarks: BookmarksModelFactory {
         }
     }
 
-    private func getRootChildren() -> Cursor<BookmarkNode> {
+    private func getRootChildren() throws -> Cursor<BookmarkNode> {
         let args: Args = [BookmarkRoots.RootID, BookmarkRoots.RootID]
         let sql = "parent = ? AND id IS NOT ?"
-        return self.getChildrenWhere(sql, args: args, includeIcon: true)
+        return try self.getChildrenWhere(sql, args: args, includeIcon: true)
     }
 
-    private func getChildren(guid: String) -> Cursor<BookmarkNode> {
+    private func getChildren(guid: String) throws -> Cursor<BookmarkNode> {
         let args: Args = [guid]
         let sql = "parent IS NOT NULL AND parent = (SELECT id FROM \(TableBookmarks) WHERE guid = ?)"
-        return self.getChildrenWhere(sql, args: args, includeIcon: true)
+        return try self.getChildrenWhere(sql, args: args, includeIcon: true)
     }
 
     func folderForGUID(guid: GUID, title: String) -> BookmarkFolder? {
-        let children = getChildren(guid)
-        if children.status == .Failure {
-            log.warning("Couldn't get children: \(children.statusMessage).")
-            return nil
-        }
+        do {
+            let children = try getChildren(guid)
+            if children.status == .Failure {
+                log.warning("Couldn't get children: \(children.statusMessage).")
+                return nil
+            }
 
-        let folder = SQLiteBookmarkFolder(guid: guid, title: title, children: children)
+            let folder = SQLiteBookmarkFolder(guid: guid, title: title, children: children)
 
-        // We add some suggested sites to the mobile bookmarks folder only.
-        if guid == BookmarkRoots.MobileFolderGUID {
-            return BookmarkFolderWithDefaults(folder: folder, sites: SuggestedSites)
+            // We add some suggested sites to the mobile bookmarks folder only.
+            if guid == BookmarkRoots.MobileFolderGUID {
+                return BookmarkFolderWithDefaults(folder: folder, sites: SuggestedSites)
+            }
+        
+            return folder
+        } catch let error as NSError {
```

Comment:
Let's put the catch closer to the `try`, so we don't have to reindent the world:

```
let children: Cursor<BookmarkNode>
do {
    children = try getChildren(guid)
} catch {
    log.error(…)
    return nil
}
```

I *think* that'll compile happily — perhaps with `var`, and it basically guards a throwing call.

We can also express this as `guard let children = try? self.getChildren(guid) else {`, which you might prefer.